sw/device/silicon_creator/lib/base/sec_mmio_unittest.cc

```diff
@@ -173,54 +173,53 @@ TEST_F(SecMmioTest, CheckCount) {
   EXPECT_EQ(ctx_->check_count, 2);
 }
 
-// Negative test cases trigger assertions, which are caugth by `ASSERT_DEATH`
-// calls. All test cases use lambda functions to wrap expectations and work
-// around issue google/googletest#1004.
+// Negative test cases trigger assertions, which are caugth by `EXPECT_DEATH`
+// calls.
 class SecMmioDeathTest : public SecMmioTest {};
 
 TEST_F(SecMmioDeathTest, Read32OrDieSimulatedFault) {
-  ASSERT_DEATH(
-      [] {
+  EXPECT_DEATH(
+      {
         EXPECT_ABS_READ32(0, 0x12345678);
         EXPECT_ABS_READ32(0, 0);
         sec_mmio_read32(0);
-      }(),
+      },
       "");
```

Comment:
Should we make this emit an appropriate error message so that we know that this is actually crashing the way we'd like it to.

Reply:
Created #10723.